builds/e2e/templates/e2e-setup.yaml

```diff
@@ -22,34 +22,14 @@ steps:
       TestRootCaPassword,
       TestBlobStoreSas
 
-- pwsh: |
-    $imageBuildId = $(resources.pipeline.images.runID)
-    $packageBuildId = $(resources.pipeline.packages.runID)
-
-    if ('$(az.pipeline.images.buildId)')
-    {
-      Write-Output '>> User supplied az.pipeline.images.buildId=$(az.pipeline.images.buildId)'
-      $imageBuildId = '$(az.pipeline.images.buildId)'
-    }
-
-    if ('$(az.pipeline.packages.buildId)')
-    {
-      Write-Output '>> User supplied az.pipeline.packages.buildId=$(az.pipeline.packages.buildId)'
-      $packageBuildId = '$(az.pipeline.packages.buildId)'
-    }
-
-    Write-Output "##vso[task.setvariable variable=imageBuildId]$imageBuildId"
-    Write-Output "##vso[task.setvariable variable=packageBuildId]$packageBuildId"
-  displayName: Override artifacts with user-supplied args
-
 - task: DownloadBuildArtifacts@0
   displayName: Get Docker image info
   inputs:
     buildType: specific
     project: $(resources.pipeline.images.projectID)
     pipeline: $(resources.pipeline.images.pipelineName)
     buildVersionToDownload: specific
-    buildId: $(imageBuildId)
+    buildId: $(resources.pipeline.images.runID)
```

Comment:
Hmm... I can imagine a couple scenario that can be impacted by this change. 
1. A dev want to run their debug artifact branch against the E2E pipeline. If I'm reading the documentation correctly, this would not allow a person to do so. 
2. From the look in "trigger" of pipeline, we are currently this is pulling the artifacts from master branch. Given the *.yaml is used as a template across 3 branches, we may run into some incompatibility between the test procedure & test artifacts?

Reply:
1. When someone manually runs this pipeline, they can still choose the artifact runs uses the "Resources" UI:
![image](https://user-images.githubusercontent.com/9042469/118847899-15bf0980-b883-11eb-9132-8fa3ca2eaac0.png)
2. The [resource trigger docs](https://docs.microsoft.com/en-us/azure/devops/pipelines/yaml-schema?view=azure-devops&tabs=schema%2Cparameter-schema#pipeline-resource) say this:

    > ...if [the resource trigger's] pipeline resource is from the same repo as the current pipeline, triggering follows the same branch and commit on which the event is raised.

    So we shouldn't be pulling the artifacts from the master branch.

Reply:
Ohhh that's super neat! Thank you for the clarification. I was thinking of Edit -> Triggers -> Yaml -> "Get source". (That was how an older pipeline get its artifacts/resources).